src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -436,6 +436,7 @@ else if (useStrictConsistency)
                          Set<InetAddressAndPort> endpointsStillReplicated = newEndpoints.endpoints();
                          // Remove new endpoints from old endpoints based on address
                          oldEndpoints = oldEndpoints.filter(r -> !endpointsStillReplicated.contains(r.endpoint()));
+                         oldEndpoints.filter(testSourceFilters);
```

Comment:
Uh, this doesn't store the result and the tests all still pass so it wasn't doing anything anyways?

Reply:
Fixed